src/data/set/intervals/default.lean

```diff
@@ -0,0 +1,92 @@
+/-
```

Comment:
I don't really understand how/why you split `intervals.lean` into `default.lean` and `basic.lean`. For instance, why is `Ico_diff_Iio` in the former instead of the latter? This should probably be explained in a file-level docstring in `default.lean`.

Reply:
I want to use `intervals.basic` in `bounded`. I'll see if I can move more lemmas to `basic` without creating a cycle. One of the reasons I can't just move `is_lub`/`is_glb` related parts to `bounded` is that some lemmas depend on `set.lattice` for `disjoint` which depends on `complete_lattice`, and I want to replace most proofs in `complete_lattice` with terms like `is_lub_Sup.*`.
I tried to split `set.complete_lattice` into `set.bounded_lattice` and `set.complete_lattice` (`disjoint` only needs an `inf_semilattice`) but this seems to create some diamonds I failed to resolve.

Reply:
Sure, I get it. What I don't get is why the lemmas in `default.lean` would create a cycle: they seem completely of the same flavor as those in `basic.lean`.

Reply:
@urkud Traditionally `default.lean` should only contain `import` statements. For example, if you write `import tactic`, this will import the "default" set of tactics, namely those imported by `tactic.default`. So I think there shouldn't be any lemmas in `default.lean`.

Reply:
I guess you can name this file `disjoint.lean` now that it only talks about disjointness of intervals.

Reply:
Done.